__tests__/users.ts

```diff
@@ -579,6 +583,72 @@ describe('query userStreaks', () => {
 
       await expectStreak(5, 5, lastViewAt);
     });
+
+    describe('streak recover query', () => {
+      const QUERY = `query StreakRecover {
+        streakRecover {
+          canDo
+          cost
+          oldStreakLength
+        }
+      }`;
+
+      it('should return recover data when user fetch query', async () => {
+        nock('http://localhost:5000').post('/e').reply(204);
+        loggedUser = '1';
+
+        const { data, errors } = await client.query(QUERY);
+        expect(errors).toBeFalsy();
+        const { streakRecover } = data;
+        expect(streakRecover).toHaveProperty('canDo');
+        expect(streakRecover.canDo).toBeFalsy();
+        expect(streakRecover).toHaveProperty('cost');
+        expect(streakRecover).toHaveProperty('oldStreakLength');
+      });
+
+      it('should disallow recover when user is not authenticated', async () => {
+        loggedUser = null;
+        const { data, errors } = await client.query(QUERY);
+        expect(errors).toBeTruthy();
+        expect(data.streakRecover).toBeNull();
+      });
+
+      it('should disallow recover when user has no streak', async () => {
+        loggedUser = '2';
+        const { data, errors } = await client.query(QUERY);
+        expect(errors).toBeFalsy();
+        expect(data.streakRecover.canDo).toBeFalsy();
```

Comment:
again `testQueryErrorCode`

Reply:
This time is different, we need to return the object but is important that `canDo === false`

Reply:
pardon, seen it wrong thought you were expecting error here, my bad